redfish-core/include/utils/hw_isolation.hpp

```diff
@@ -335,5 +342,297 @@ inline void
+                                            << "message registry to add "
+                                            << "in the condition";
+                                        messages::internalError(aResp->res);
+                                        break;
+                                    }
+
+                                    // Prepare MessageArgs as per defined in the
+                                    // MessageRegistries
+                                    std::vector<std::string> messageArgs{
+                                        *msgPropVal};
+
+                                    // Fill the "msgPropVal" as reason
+                                    std::string message = msgReg->message;
+                                    int i = 0;
+                                    for (const std::string& messageArg :
+                                         messageArgs)
+                                    {
+                                        std::string argIndex =
+                                            "%" + std::to_string(++i);
+                                        size_t argPos = message.find(argIndex);
+                                        if (argPos != std::string::npos)
+                                        {
+                                            message.replace(argPos,
+                                                            argIndex.length(),
+                                                            messageArg);
+                                        }
+                                    }
+                                    // Severity will be added based on the event
+                                    // object property
+                                    condition["Message"] = message;
+                                    condition["MessageArgs"] = messageArgs;
+                                    condition["MessageId"] =
+                                        "OpenBMC.0.2.HardwareIsolationReason";
+                                }
+                                else if (property.first == "Severity")
+                                {
+                                    const std::string* severity =
+                                        std::get_if<std::string>(
+                                            &property.second);
+                                    if (severity == nullptr)
+                                    {
+                                        BMCWEB_LOG_ERROR
+                                            << "Failed to get the Severity "
+                                            << "from object: "
+                                            << hwStatusEventObj;
+                                        messages::internalError(aResp->res);
+                                        break;
+                                    }
+
+                                    if (*severity ==
```

Comment:
typically lines 589 - 618 are a separate function upstream

Reply:
Typically upstream, for example: https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/43761/15/redfish-core/lib/systems.hpp

Reply:
I like our downstream commits to match upstream as much as possible for easier maintainability and one day I hope a lot of these downstream patches do go upstream

Reply:
Hmm got it, I think I have addressed that [here](https://github.com/ibm-openbmc/bmcweb/pull/171/commits/56de3b4047b4648989473ddf5188737dd0a96384#diff-2b15c834283a368213b30e0c0e8e913e313e61bd8060d94a79d434b99f1e127eR353) but, with some minor changes to align with hardware isolation status use case i.e
- Returning flag if unsupported severity because we are throwing an internal error if any of attribute set is failed.
- Passing object path include in trace for better debug.
- Passing property path so that we can extend future while adding more than one condition.